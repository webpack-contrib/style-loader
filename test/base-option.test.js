import {
  compile,
  getCompiler,
  getErrors,
  getWarnings,
  runInJsDom,
} from "./helpers/index";

describe('"base" option', () => {
  it("should work", async () => {
    expect.assertions(3);

    const compiler = getCompiler("./simple.js", { base: 1000 });
    const stats = await compile(compiler);

    runInJsDom("main.bundle.js", compiler, stats, (dom) => {
      expect(dom.serialize()).toMatchSnapshot("DOM");
    });

    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });
});
