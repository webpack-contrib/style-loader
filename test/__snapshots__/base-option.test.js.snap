// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"base" option should work: DOM 1`] = `
"<!DOCTYPE html><html><head>
  <title>style-loader test</title>
  <style id=\\"existing-style\\">.existing { color: yellow }</style>
<style>body {
  color: red;
}
</style><style>h1 {
  color: blue;
}
</style></head>
<body>
  <h1>Body</h1>
  <div class=\\"target\\"></div>
  <iframe class=\\"iframeTarget\\"></iframe>


</body></html>"
`;

exports[`"base" option should work: errors 1`] = `Array []`;

exports[`"base" option should work: warnings 1`] = `Array []`;
